pandas/core/dtypes/common.py

```diff
@@ -1789,7 +1789,7 @@ def _coerce_to_dtype(dtype):
         ordered = getattr(dtype, 'ordered', False)
         dtype = CategoricalDtype(categories=categories, ordered=ordered)
     elif is_datetime64tz_dtype(dtype):
-        dtype = DatetimeTZDtype(dtype)
+        dtype = DatetimeTZDtype.construct_from_string(dtype)
```

Comment:
this is odd that you needed to change. the dtype is by definition a valid dtype already, so its *not* a string

Reply:
Looks like we have a test that calls this directly.

```python
176         def test_coerce_to_dtype(self):
177  ->         assert (_coerce_to_dtype('datetime64[ns, US/Eastern]') ==
178                     DatetimeTZDtype('ns', 'US/Eastern'))
179             assert (_coerce_to_dtype('datetime64[ns, Asia/Tokyo]') ==
180                     DatetimeTZDtype('ns', 'Asia/Tokyo'))
```

I'll skip that test and see if we have any real use cases passing a string here.

Reply:
Well, this should be easy since we don't seem to *use* `_coerce_to_dtype` outside of tests :)

Reply:
hmm, let's blow that away as very duplicative code